plugins/modules/instance.py

```diff
@@ -851,6 +864,23 @@ def _update_instance(self) -> None:
         if should_update:
             self._instance.save()
 
+        ipv4_length = len(self.module.params.get("additional_ipv4") or [])
+
+        if self.module.params.get("private_ip"):
+            if ipv4_length != len(getattr(self._instance, "ipv4")) - 2:
+                self.fail(
+                    "failed to update instance {0}:additional_ipv4 is a non-updatable field".format(
+                        self._instance.label
+                    )
+                )
+        else:
+            if ipv4_length != len(getattr(self._instance, "ipv4")) - 1:
+                self.fail(
+                    "failed to update instance {0}:additional_ipv4 is a non-updatable field".format(
+                        self._instance.label
+                    )
+                )
+
```

Comment:
Since they would throw the same error message, can you try to combine these 2 if-else branches?
You can try something like:
```python
min_ips = 2 if self.module.params.get("private_ip") else 1
if ipv4_length != len(getattr(self._instance, "ipv4")) - min_ips:
    ...
```

Reply:
Thanks! Took care of this.